operators.py

```diff
@@ -2726,12 +2734,26 @@ def activate_sketch(context, index, operator):
         if not sk:
             operator.report({"ERROR"}, "Invalid index: {}".format(index))
             return {"CANCELLED"}
-
+        
         space_data.show_object_viewport_curve = False
         space_data.show_object_viewport_mesh = False
+        
+        #Align view to normal of wp
+        n = Vector((sk.wp.normal.x,sk.wp.normal.y,-sk.wp.normal.z))
+        v = Vector((0,0,1))        
+        quat = n.rotation_difference(v)
+        context.region_data.view_rotation = quat
+        context.region_data.view_location = sk.wp.p1.location #lookat this point
+        context.region_data.view_distance = 6 # from this far away
+
     else:
         space_data.show_object_viewport_curve = True
         space_data.show_object_viewport_mesh = True
+        
+        #Reset view
+        context.region_data.view_rotation = (Euler((0.7854,0,0.7854*3),'XYZ').to_quaternion())
```

Comment:
Generally it's better to avoid using magic numbers and give them a symbolic name. In this case it could be something like this: PI_DIV_4 = math.pi / 4